### PR TITLE
firmware update 

### PR DESCRIPTION
**Pull request by mroz22**
first shot of firmware update in suite. It works now, but only for "happy path" for model. I have a bunch of things to discuss.

### CODE REVIEW

**Comment by mroz22** [comment]
slowly comming to rc phase.

There is still bunch of things to do, but probably not in this commit:
- [ ] refactor notifications from wallet/ to suite/ (notification is a common thing) and use them for indicating errors resulting from fw update.
- [ ] resolve what happens if device disconnects (it should not show connect device page)
- [ ] resolve reconnecting for T1 (not tested or tried yet)
- [ ] resolve reconnecting of different device than device previously connected 
- [ ] improve test coverage
- [ ] try rolling update 
...
 
lot of stuff

File: packages/suite/src/views/suite/firmware/index.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    };
+
+    return (
+        <Wrapper>
+            {/* todo: use proper notifications, leaving it here as a starting point for next iteration */}
+            <WalletNotifications />
+            <Top>
+                <TitleHeader>
+                    {/* todo: this does not appear right now, it displays ConnectDevice page. */}
+                    {getStatus() === 'no-device'
```
Comment (slowbackspace) [inline packages/suite/src/views/suite/firmware/index.tsx]: Screenshot 2019-08-27 at 14 06 43

They are not. I am calling for reenabling this rule and banning nested ternary 🚨

**Review by slowbackspace — CHANGES_REQUESTED** [review]

File: packages/suite/src/views/suite/firmware/index.tsx (hunk 1 above)
Comment (mroz22) [reply packages/suite/src/views/suite/firmware/index.tsx]: HAHA :D Well. For example in this case, I can imagine a function, something like `getHeadingTextForStatus(status)`. But what if I have quite complex tree to render based on many conditions. Of course the tree might be binary, but nesting will become quite deep then. Of course, I could create a fucntion `getComponentForStatus(status)` and return entire component based on given conditions. If this is what you like more, why not.

Comment (slowbackspace) [reply packages/suite/src/views/suite/firmware/index.tsx]: Well, I don't fully understand your usecase, but there are whole languages without a ternary operator. I am sure you can find a way to write your code without NESTED ternary op 😀 They are also disabled in eslint by default, for a good reason imo. 
http://golang.org/doc/faq#Does_Go_have_a_ternary_form

**Review by slowbackspace — APPROVED** [review]

**Pull request merged**